src/core/execute.c

```diff
@@ -7281,7 +7347,7 @@ ExecSharedRuntime* exec_shared_runtime_destroy(ExecSharedRuntime *rt) {
         if (rt->tmp_dir && !streq(rt->tmp_dir, RUN_SYSTEMD_EMPTY)) {
                 log_debug("Spawning thread to nuke %s", rt->tmp_dir);
 
-                r = asynchronous_job(remove_tmpdir_thread, rt->tmp_dir);
+                r = asynchronous_job(rm_rf_thread, rt->tmp_dir);
                 if (r < 0)
                         log_warning_errno(r, "Failed to nuke %s: %m", rt->tmp_dir);
                 else
```

Comment:
maybe wrap these 4 lines into a new helper call. after all you have the same 4 lines 3 times now